models/stg_apple_search_ads__ad_group_history.sql

```diff
@@ -23,7 +23,7 @@ final as (
     
     select 
         _fivetran_synced,
-        modification_time as modified_at,
+        modification_time as modified_at, --We should check to see if we will need to add logic to get the most recent ad_group
```

Comment:
Typically in these `history` tables, we need to account for records that change. See how we did this in our Snapchat Ads package: 
https://github.com/fivetran/dbt_snapchat_ads_source/blob/main/models/stg_snapchat__ad_squad_history.sql#L34-L37

I have a suspicion we will need to apply similar logic here. We can test this and see if the result replicates into a new record. Or if the modified time simply updates.

Reply:
Ah, I understand that logic! In the case of snapchat, is that to remove duplicates? Or to get the latest set of data for a specific record?

I think the `most_recent` record would make sense if the records are duplicated. But it looks like at least for the `campaign_history` table that new records are created when there are updates to a campaign. Although we only have unique records for the `keyword_history` and `ad_group_history`, I have a hunch that these tables will also follow the same logic.

A use case that I'm thinking:
Specifically for the `campaign_history` table, it looks like this table records changes to the record. E.g. when a campaign status goes from `ENABLED` -> `PAUSE`. It's more like a campaign "events" table if that makes sense. I think in this case, it'd be valuable to have all the records and not just the most recent - especially if, for example, users want to calculate the difference in time between a campaign "TOTAL_BUDGET_EXHAUSTED" and when it was supposed to end. You can see this example in `dbt_package_testing.apple_search_ads` (there's only 3 records).

Reply:
I'm almost wondering if there should be a 1. campaign "look up" table and 2. campaign "events" table. This logic would extend to ad_group_history & keyword_history as well. 

A little more detail (which can be extended to the other two tables):
1. Campaign look up table should be the "cleaned" table with soft deleted records removed, and previous events removed.
2. Campaign "events" table should have all records, including soft deleted records & previous changes to that campaign.